go/vt/vtgate/planbuilder/abstract/queryprojection.go

```diff
@@ -57,12 +60,42 @@ type (
 		Inner         sqlparser.Expr
 		WeightStrExpr sqlparser.Expr
 
-		// This is to add the distinct function expression in grouping column for pushing down but not be to used as grouping key at VTGate level.
-		// Starts with 1 so that default (0) means unassigned.
-		DistinctAggrIndex int
+		// The index at which the user expects to see this column. Set to nil, if the user does not ask for it
+		InnerIndex  *int
+		aliasedExpr *sqlparser.AliasedExpr
```

Comment:
Could we also add a comment here for `aliasedExpr` which is set only if the group by field is present in the select list.